app/models/agents/rss_agent.rb

```diff
@@ -113,7 +113,7 @@ def check_urls(urls)
 
       created_event_count = 0
       sort_events(new_events).each.with_index do |event, index|
-        entry_id = event.payload[:id]
+        entry_id = event.payload[:update_id]
```

Comment:
I haven't tried this yet, but I think one way to achieve backwards compatibility might be something like this [line 117](https://github.com/irfancharania/huginn/blob/936077e3c88fdde2393892effa6e39e67a858163/app/models/agents/rss_agent.rb#L117)

``` ruby

entry_id = event.payload[:id]
entry_update_id = event.payload[:update_id]
        if check_and_track(entry_update_id) || check_and_track(entry_id)
```

New entries would match the first case, older entries would match the second case
I'll try it out later and report back